### PR TITLE
[Sketcher] cursor display unit awareness

### PR DESCRIPTION
**Pull request by WandererFan**
https://www.forum.freecadweb.org/viewtopic.php?f=3&t=71354

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7420">pipeline status</a> for feature branch [PR_7420](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7420). Pipeline [627225818](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/627225818) was triggered at [c45d75f](https://github.com/FreeCAD/FreeCAD/pull/7420/commits/c45d75f71d5ff1abff1d78d4426cebe4970c2f48). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/Sketcher/Gui/CommandSketcherTools.cpp [hunk 1]
```diff
@@ -1125,9 +1125,13 @@ class DrawSketchHandlerCopy: public DrawSketchHandler
...
             }
 
-            SbString text;
-            text.sprintf(" (%.1f, %.1fdeg)", length, angle * 180 / M_PI);
-            setPositionText(endpoint, text);
+            if (showCursorCoords()) {
+                SbString text;
+                std::string lengthString = lengthToDisplayFormat(length, 1);
+                std::string angleString = angleToDisplayFormat(angle * 180.0 / M_PI, 1);
+                text.sprintf(" (%s, %.s°)", lengthString.c_str(), angleString.c_str());
```
Comment (0penBrain) [inline src/Mod/Sketcher/Gui/CommandSketcherTools.cpp:1132]: Extra '°' character ?

Comment (WandererFan) [reply src/Mod/Sketcher/Gui/CommandSketcherTools.cpp:1132]: Yes, thanks.  Last one I hope.

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7420">pipeline status</a> for feature branch [PR_7420](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7420). Pipeline [628007194](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/628007194) was triggered at [dfdd3d9](https://github.com/FreeCAD/FreeCAD/pull/7420/commits/dfdd3d9fff450162420bffcf8c34c70863b00bee). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**